src/main/java/de/tum/in/www1/artemis/domain/lti/Lti13AgsClaim.java

```diff
@@ -20,33 +20,38 @@ public class Lti13AgsClaim {
      *
      * @param idToken to be parsed
      * @return an Ags-Claim if one was present in idToken.
+     * @throws IllegalStateException if the AGS claim is present but malformed or cannot be processed.
      */
     public static Optional<Lti13AgsClaim> from(OidcIdToken idToken) {
         if (idToken.getClaim(Claims.AGS_CLAIM) == null) {
             return Optional.empty();
         }
 
-        JsonObject agsClaimJson = JsonParser.parseString(idToken.getClaim(Claims.AGS_CLAIM).toString()).getAsJsonObject();
-
-        Lti13AgsClaim agsClaim = new Lti13AgsClaim();
-        JsonArray scopes = agsClaimJson.get("scope").getAsJsonArray();
-
-        if (scopes == null) {
-            return Optional.empty();
-        }
-
-        if (scopes.contains(new JsonPrimitive(Scopes.AGS_SCORE))) {
-            agsClaim.setScope(Collections.singletonList(Scopes.AGS_SCORE));
-        }
-
-        JsonElement lineItem = agsClaimJson.get("lineitem");
-        if (lineItem != null) {
-            agsClaim.setLineItem(lineItem.getAsString());
+        try {
+            JsonObject agsClaimJson = new Gson().toJsonTree(idToken.getClaim(Claims.AGS_CLAIM)).getAsJsonObject();
+            Lti13AgsClaim agsClaim = new Lti13AgsClaim();
+            JsonArray scopes = agsClaimJson.get("scope").getAsJsonArray();
+
+            if (scopes == null) {
+                return Optional.empty();
+            }
+
+            if (scopes.contains(new JsonPrimitive(Scopes.AGS_SCORE))) {
+                agsClaim.setScope(Collections.singletonList(Scopes.AGS_SCORE));
+            }
+
+            JsonElement lineItem = agsClaimJson.get("lineitem");
+            if (lineItem != null) {
+                agsClaim.setLineItem(lineItem.getAsString());
+            }
+            else {
+                agsClaim.setLineItem(null);
+            }
+            return Optional.of(agsClaim);
```

Comment:
The intention behind returning Optional.empty() for a null scopes is to handle cases where the AGS claim might not include a scope field, which doesn't necessarily indicate a malformed claim. In such scenarios, we treat the absence of scope as a valid state, hence the choice to return an empty Optional rather than throwing an exception. This approach aligns with our current handling of optional claim fields, ensuring the method remains flexible for various AGS claim structures.